src/code/z_fireobj.c

```diff
@@ -168,28 +168,21 @@ void FireObj_Draw(GlobalContext* globalCtx, FireObj* fire) {
     }
 }
 
-#ifdef NON_EQUIVALENT
-// Accesses data incorrectly
-void FireObj_InitLight(GlobalContext* globalCtx, FireObjLight* light, u8* param_3, Vec3f* pos) {
-    FireObjLightParams* objectParams = &sFireObjLightParams[*param_3];
+void FireObj_InitLight(GlobalContext* globalCtx, FireObjLight* light, u8* paramsIndex, Vec3f* pos) {
+    FireObjLightParams* objectParams = &sFireObjLightParams[*paramsIndex];
 
-    Lights_PointGlowSetInfo(&light->lightInfo, pos->x, pos->y, pos->z, objectParams->primColor.g,
-                            objectParams->primColor.b, objectParams->primColor.a, objectParams->primColor.r);
+    Lights_PointGlowSetInfo((LightInfo*)&light->lightInfo, pos->x, pos->y, pos->z, objectParams->color.r,
```

Comment:
This is a weird cast.
Looking at the structs, seems like `LightInfo` and `LightInfoPositional` are very similar. I wonder if we have those structs wrong...
I'm not saying you should try to clean up those structs in this PR (it would probably be outside the scope of this PR), just mentioning it. Somebody should look at it tho.

Reply:
Positional is new to MM, so could be either way.

Reply:
So looking it over, and per the discussion on discord, I changed the `lightInfo` variable in `FireObj` from a `LightInfoPositional` to a `LightInfo`. Also deleted the `LightInfoPositional` struct as it seems redundant with the `LightInfo` struct.